### DISCUSSION TITLE
为什么固件的程序（ElectronBot-fw）里面既有main.c也有main.cpp呀？新手不太懂，希望有人能解答一下，谢谢

### DISCUSSION THREAD

**Discussion by rookielittlep** [peng-zhihui/ElectronBot#195] [Q&A]
为什么固件的程序（ElectronBot-fw）里面既有main.c也有main.cpp呀？新手不太懂，希望有人能解答一下，谢谢

**Comment by carveybunt** [comment]
main.c 是C语言的源文件；main.cpp 是C++的源文件。

**Comment by Kitem** [comment]
您好，我是许焕挺，已收到您的来信，谢谢。

**Reply by carveybunt** [reply]
你的相关耗材都是从哪里购买的呀？我也想复刻一个

**Comment by rookielittlep** [comment]
基本都是在淘宝上买的，你可以看一下b站飞起小鹏的视频



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2022年7月30日(周六) 中午12:07
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [peng-zhihui/ElectronBot] 为什么固件的程序（ElectronBot-fw）里面既有main.c也有main.cpp呀？新手不太懂，希望有人能解答一下，谢谢 (Discussion #195)




 
你的相关耗材都是从哪里购买的呀？我也想复刻一个

**Reply by carveybunt** [reply]
好的，多谢

**Comment by Eilgnaw** [comment]
看一下 Core/Src/main.c 里的 MX_FREERTOS_Init();
